app/application/services/analyze_skills.rb

```diff
@@ -100,66 +101,68 @@ def store_query_to_db(input)
       end
 
       # Pass to response object
+      # :reek:UncommunicativeVariableName for rescued error
       def to_response_object(input)
         result_response = Response::Result.new(input[:query], input[:jobs], input[:skills], input[:salary_dist])
         Success(Response::ApiResult.new(status: :ok, message: result_response))
       rescue StandardError => e
         Failure(Response::ApiResult.new(status: :internal_error, message: "Fail to map to response object: #{e}"))
       end
 
-      # ------ UTILITIES ------ #
-
-      # search corresponding jobs in database first,
-      # or request it through JobMapper
-      def search_jobs(input)
-        query = input[:query]
-        if Repository::QueriesJobs.query_exist?(query)
-          Repository::QueriesJobs.find_jobs_by_query(query)
-        else
-          request_jobs_and_update_database(query)
+      # An utility class that handle job processing in the service
+      class Utility
```

Comment:
I would recommend this `Utility` class be placed into a new file.

Reply:
Fixed in [PR#11](https://github.com/KeroKnows/s-killer-api/pull/11/commits/0b57c53dbc5efaaac222d9c00b311469e63e1847). Thanks!